packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -867,6 +873,10 @@ export class GarbageCollector implements IGarbageCollector {
                 }
                 if (newDeletedNodes.length > 0) {
                     // Call container runtime to delete these nodes and add deleted nodes to this.deletedNodes.
+                    const deletedNodes = this.runtime.deleteUnusedNodes(newDeletedNodes);
```

Comment:
On a separate note, I was thinking if we should validate that `refreshLatestSummary` is called for a reference sequence number that is newer than the last GC run's reference sequence number. Otherwise, we may move back in time and have more deleted nodes than expected. For example, the last GC run may have deleted nodes that are not deleted in this snapshot we are refreshing from and we are left in an inconsistent state.

Reply:
If this happened, it may also break some validation that you are planning to add in `removeDeletedNodesFromGCState`.

Reply:
This comment and the one below makes me think we should do testing based on refresh latest snapshot. I'll add more tests.

Reply:
I added refresh tests to `gcLoadingFromOlderSummaries.spec.ts`. Take a look at those.

Reply:
We already see this happening in the stress tests :-(.
https://dev.azure.com/fluidframework/internal/_workitems/edit/3324

Reply:
Yeah, I'm running into this issue right now with the code changes you suggested. I think it would be highly advantageous to simplify the logic we are running into one `setGCStateFromSnapshot` function, and then it wouldn't matter if we were loading from a snapshot, refreshing an ack from a snapshot, etc. That way we would get to the same baseGCState regardless of which snapshot we loaded from, and then the logic for when we are setting our state from a snapshot is isolated to a higher layer

Reply:
You are running into this issue due to which code changes?
Also, this function is what you are talking about. It's called during loading from base snapshot and during refresh from snapshot.
I would also consolidate the sweep plus related state updated into a single function so that only one function needs to be called making the call sites simpler.

Reply:
I think I will leave this part for a separate PR as we do not have consensus yet on what we want to do for refresh from snapshot and the different edge cases we may run into.